src/core/qgsgnsspositioninformation.cpp

```diff
@@ -0,0 +1,126 @@
+/***************************************************************************
+  qgsgnsspositioninformation.cpp - QgsGnssPositionInformation
+ ---------------------
+ begin                : 1.12.2020
+ copyright            : (C) 2020 by David Signer
+ email                : david (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsgnsspositioninformation.h"
+
+#include <QCoreApplication>
+#include <QTime>
+#include <QIODevice>
+#include <QStringList>
+#include <QFileInfo>
+
+#include "qgsnmeaconnection.h"
+#include "qgslogger.h"
+
+
+QgsGnssPositionInformation::QgsGnssPositionInformation(double latitude, double longitude, double elevation, double speed, double direction, const QList<QgsSatelliteInfo> &satellitesInView, double pdop, double hdop, double vdop, double hacc, double vacc,
```

Comment:
Is this a QGIS (Qgs) class or a QField class?

Reply:
I thought about that. It's not a QGIS class but maybe something we could suggest for 4.0 to QGIS, that's why I named it like this. But I'm open to rename it to `GnnsPositionInformation`

Reply:
I would keep it as GnssPositionInformation for now.
Regarding moving to QGIS, is it something that needs to wait for QGIS 4 or do you see a way to propose it already now to allow for a smooth transition?

Reply:
Not really. And if so it can be changed easily. I do prefer `GnssPositionInformation` as well. What is the purpose btw. of Qgs prefixed classes like `QgsGpkgFlusher` and `QgsGeometryWrapper` etc?

Reply:
GpkgFlusher: accident
GeomWrapper: likely because it wraps a QgsGeometry

Reply:
Fair enough